src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftBuilder.java

```diff
@@ -107,7 +107,7 @@ default void waitOnBuild(IClient client, long startTime, String bldId, TaskListe
 			if (Boolean.parseBoolean(getVerbose(overrides)))
 				listener.getLogger().println("\nOpenShiftBuilder bld state:  " + bldState);
 			
-			if (follow) {
+			if (follow && bldState.equals("Running") && stop == null) {
```

Comment:
interesting - somehow I thought I remembered that the accept was blocking, but it has been a while; also, I don't recall the checking for stop == null coming up when discussing usage with the restclient folks;

that said, while it wasn't abundant when I saw it, I had noticed *some* duplication of logs recently

Can I assume you rebuilt the plugin and tried these changes out ?  If so, IGTM

thanks @jim-minter !!!